### ISSUE TITLE
Training and inference discussion for active learning round 3

### ISSUE THREAD

**Issue by rohanbanerjee** [sct-pipeline/fmri-segmentation#40]
Continuation from the previous round of training: https://github.com/sct-pipeline/fmri-segmentation/issues/35

#### What is the round 3 model
The model which was fine-tuned on the manually corrected segmentations as per the QCs mentioned in #38 is the `round 3` model. A total of 40 images were added in the training of this model since we fine-tuned the previously trained `round 2` model. 

A list of subjects used for the `fine-tuning` is below:  [finetuning.yml](https://github.com/sct-pipeline/fmri-segmentation/files/15040610/qc_fail.yml.zip)

The config (containing preprocessing, hyperparameters) for nnUNetv2 training is: [plans.json](https://github.com/sct-pipeline/fmri-segmentation/files/15079506/plans.json)

After the training was completed, I ran inference on the rest of the images whose segmentations have to be included in the consequent rounds of training (186 images), below is the QC. 50 subjects from these images will be chosen and included in the `round 4` of training:

[qc_round3_inference.zip](https://github.com/sct-pipeline/fmri-segmentation/files/15079536/qc_round3_inference.zip)

The steps to reproduce the above QC results (/run inference) are the following:
1. Clone this repos
2. Get the data: XXX
3. `cd fmri-segmentation`
4. Download the model weights (the whole folder) from the link: https://drive.google.com/drive/folders/1AW_myyHR-hA7xF_ckbxV45RXXRsKnlZC?usp=sharing
5. For inference, please follow instructions under the inference directory
 

Next steps:
- [x] Add inference QCs in this comment
- [x] Choose and correct 50 segmentations and include it in the training model 4 round along with the ones that have good segmentation after inference.
- [ ]  Get metric by running evaluation on the `held-out test set` (#33)

**Comment by rohanbanerjee** [comment]
The list of 50 subjects chosen for manual corrections is below:
[qc_fail.yml.zip](https://github.com/sct-pipeline/fmri-segmentation/files/15125623/qc_fail.yml.zip)

The QC report for the segmentations manually corrected by me for the above 50 subjects is below:
[qc_round_3_corrected.zip](https://github.com/sct-pipeline/fmri-segmentation/files/15125633/qc_round_3_corrected.zip)

**Comment by jcohenadad** [comment]
improper shape of the cord (look carefully at the boundaries when making manual correction):
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/8efe8816-6b10-4908-bb1e-40a2dfe8fcdb)

underseg:
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/5ab1f420-6845-4327-942a-1ee01dfd7483)

incorrect:
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/66606f9b-e4d8-4e66-b6a1-c8c75d52ca9e)

I'll stop doing the rest of the QC-- pls make sure this is correct before sending it to me 

Here's what i did so far: [qc_flags.json](https://github.com/sct-pipeline/fmri-segmentation/files/15131063/qc_flags.json)

**Comment by rohanbanerjee** [comment]
Closing the issue since the round 3 training was successfully completed (including running inference and manual correction)
